source/Octopus.Tentacle.Client/Scripts/HalibutClientExceptionExtensionMethods.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Halibut;
+
+namespace Octopus.Tentacle.Client.Scripts
+{
+    public static class HalibutClientExceptionExtensionMethods
+    {
+        public static bool IsHalibutWrappedConnectingRequestCancelledException(this HalibutClientException? ex)
+        {
+            if (ex is null)
+            {
+                return false;
+            }
+
+            if (ex.Message.Contains("The Request was cancelled while Connecting"))
```

Comment:
I suspect you've explored all this, but is there not a type-safe way to detect this? Or even a string in the message (or other property) that has a text-based version of the type instead?

Possibly just wishful thinking 🙂

Reply:
Yeh this isn't nice code!! I'll revisit. At a minimum, this can be moved over to Halibut and tested to make sure the behaviour doesn't change. I'll see if there is an existing way to surface something more type safe!

Reply:
I'll leave this as is and revisit with a better solution in another PR. We have tests that will capture if this isn't working as expected so it shouldn't be the source of bugs.